From 3ed482724595535d2c8688175d062e93a0540d69 Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Thu, 18 Mar 2021 17:03:01 -0700
Subject: [PATCH 1786/1851] drm: xlnx: zynqmp_dpsub: Pass pcm auxdata when
 populating sound nodes

commit a9a726e74c567ef911929f54db892151695b5b66 from
https://github.com/Xilinx/linux-xlnx.git

The DP sound nodes are being populated without any auxdata which creates
sysfs files with the default device tree name. This does not match the
modified names from the DP snd pcm driver. So, pass an auxdata table
with matching names from the DP snd pcm driver.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Acked-by: Varunkumar Allagadapa <varunkumar.allagadapa@xilinx.com>
Acked-by: Jianqiang Chen <jianqiang.chen@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_dpsub.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_dpsub.c b/drivers/gpu/drm/xlnx/zynqmp_dpsub.c
index 7197547c9eeb..6ee063294e3f 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_dpsub.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_dpsub.c
@@ -29,6 +29,9 @@
 #include "zynqmp_dp.h"
 #include "zynqmp_dpsub.h"
 
+#define DP_PCM_NAME_0 "zynqmp_dp_snd_pcm0"
+#define DP_PCM_NAME_1 "zynqmp_dp_snd_pcm1"
+
 static int
 zynqmp_dpsub_bind(struct device *dev, struct device *master, void *data)
 {
@@ -58,6 +61,12 @@ static const struct component_ops zynqmp_dpsub_component_ops = {
 	.unbind	= zynqmp_dpsub_unbind,
 };
 
+static struct of_dev_auxdata zynqmp_dpsub_auxdata_lookup[] = {
+	OF_DEV_AUXDATA("xlnx,dp-snd-pcm0", 0, DP_PCM_NAME_0, NULL),
+	OF_DEV_AUXDATA("xlnx,dp-snd-pcm1", 0, DP_PCM_NAME_1, NULL),
+	{ /* end of table */ }
+};
+
 static int zynqmp_dpsub_probe(struct platform_device *pdev)
 {
 	struct zynqmp_dpsub *dpsub;
@@ -91,7 +100,8 @@ static int zynqmp_dpsub_probe(struct platform_device *pdev)
 	of_reserved_mem_device_init(&pdev->dev);
 
 	/* Populate the sound child nodes */
-	ret = of_platform_populate(pdev->dev.of_node, NULL, NULL, &pdev->dev);
+	ret = of_platform_populate(pdev->dev.of_node, NULL,
+				   zynqmp_dpsub_auxdata_lookup, &pdev->dev);
 	if (ret) {
 		dev_err(&pdev->dev, "failed to populate child nodes\n");
 		goto err_rmem;
-- 
2.31.1

